### PR TITLE
MM-58436: Notifications review feedback

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
This implements a number of pieces of design review feedback, most notably:
* Eliminate the footer on notification messages and just link inline for reduced clutter
* Support group chat names (called "topics")
* Consistent terminology usage

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-58436

### CODE REVIEW

File: assets/account-connected/index.html [hunk 1]
```diff
@@ -6,11 +6,11 @@
...
       }, 1);
 
-
       function closeInfoPage() {
-        document.getElementById("close-window").classList.remove("msteams-sync-hidden");
-        document.getElementById("info-window").classList.add("msteams-sync-hidden");
         window.close();
+        document.getElementById("close-window-failed").classList.remove("msteams-sync-hidden");
+        document.getElementById("close-window").classList.add("msteams-sync-hidden");
+        document.getElementById("close-window-icon").classList.add("msteams-sync-hidden");
```
Comment (lieut-data) [inline assets/account-connected/index.html:13]: Do this after `window.close` so there's no flash if we're successful in closing. Also hide the `X` too.

File: assets/account-connected/index.html [hunk 2]
```diff
@@ -144,6 +144,7 @@
       .msteams-sync-body .close-button__text {
         background-color: rgba(28, 88, 217, 0.08);
         color: #1c58d9;
+        cursor: pointer;
```
Comment (lieut-data) [inline assets/account-connected/index.html:147]: Show the right cursor icon for a button.

File: server/api.go [hunk 3]
```diff
@@ -407,7 +408,7 @@ func (a *API) connectionStatus(w http.ResponseWriter, r *http.Request) {
 }
 
 func (a *API) accountConnectedPage(w http.ResponseWriter, r *http.Request) {
-	message := "You are now connected."
+	message := "Your account is now connected to MS Teams."
```
Comment (lieut-data) [inline server/api.go:411]: Be explicit about what we're connected to.

File: server/api.go [hunk 4]
```diff
@@ -724,7 +725,17 @@ func (a *API) handleSyncNotificationsWelcomeMessage(originInfo string, mmUserID,
 				a.p.API.LogWarn("Unable to update post", "post", postID, "error", appErr.Error())
 			}
 		} else {
-			a.p.GetAPI().DeleteEphemeralPost(mmUserID, postID)
+			// Update the original connection prompt and remove any calls to action.
```
Comment (lieut-data) [inline server/api.go:728]: Make sure the user comes back to Mattermost with an ephemeral message explaining what happened. We also send a welcome message from the bot itself.

File: server/api.go [hunk 5]
```diff
@@ -1135,14 +1146,8 @@ func (a *API) enableNotifications(w http.ResponseWriter, r *http.Request) {
...
 
-	attachments := post.Attachments()
-	if len(attachments) == 1 && len(attachments[0].Actions) == 1 {
-		attachments[0].Actions[0].Disabled = true
-		attachments[0].Actions[0].Name = "Notifications enabled!"
-		post.SetProps(map[string]interface{}{
-			"attachments": attachments,
-		})
-	}
+	post.Message = "You will now start receiving notifications from chats or group chats in Teams. To change this setting, open your user settings or run `/msteams notifications`"
```
Comment (lieut-data) [inline server/api.go:1149]: Basically make the welcome disappear in favour of the explanation of what was chosen.

File: server/api.go [hunk 6]
```diff
@@ -1153,3 +1158,48 @@ func (a *API) enableNotifications(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "unable to encode the response", http.StatusInternalServerError)
 	}
 }
+
+func (a *API) dismissNotifications(w http.ResponseWriter, r *http.Request) {
```
Comment (lieut-data) [inline server/api.go:1162]: Support dismissing the welcome message without accepting the option.

File: server/bot_messages.go [hunk 7]
```diff
@@ -113,8 +113,8 @@ func (p *Plugin) SendWelcomeMessageWithNotificationAction(userID string) error {
 
 func (p *Plugin) makeWelcomeMessageWithNotificationActionPost() *model.Post {
 	msg := []string{
-		"**Welcome to the MS Teams integration!**",
-		"Enable notifications to get notified about any mentions you get in direct messages or group messages in MS Teams directly within Mattermost.",
+		"**Notifications from chats and group chats**",
```
Comment (lieut-data) [inline server/bot_messages.go:116]: Tweaks.

**Comment by lieut-data** [comment]
@cwarnermm, I'd love to pick your brain on some of the text in this new "notifications-only" approach to the MS Teams plugin. What's the best way for me to consolidate the changes and show case them your way?

**Comment by cwarnermm** [comment]
@lieut-data - If there's a test environment available, that may be the easiest way to see the user experience end-to-end and provide input.

File: server/bot_messages.go [hunk 8]
```diff
@@ -23,6 +23,10 @@ func (p *Plugin) botSendDirectPost(userID string, post *model.Post) error {
 	post.ChannelId = channel.Id
 	post.UserId = p.botUserID
 
+	// Force posts from the bot to render the user profile icon each time instead of collapsing
+	// adjacent posts. This helps draw attention to each individual post.
+	post.AddProp("from_webhook", "true")
```
Comment (JulienTant) [inline server/bot_messages.go:28]: It's neat that we can do that, but also kind of ugly that we have to do it this way 😅

**Comment by lieut-data** [comment]
@cwarnermm, I've pushed all the outstanding changes to https://msteams-sync-test.test.mattermost.cloud. Maybe we can review in our planned 1:1?

**Comment by lieut-data** [comment]
@cwarnermm, heads up that I'm going to merge this alongside some other PRs and we can evaluate the whole thing together tomorrow :)

**Pull request merged**